src/Layout/ErrorBoundary/index.tsx

```diff
@@ -55,50 +65,137 @@ export class ErrorBoundary extends React.PureComponent<Props, State> {
       error,
       errorInfo,
     });
+
+    if (this.testResourceNotFound(error)) {
+      this.setState({
+        shouldReload: true,
+      });
+      this.startCountdown();
+    }
+  }
+
+  public componentWillUnmount(): void {
+    this.toDispose.dispose();
+  }
+
+  private testResourceNotFound(error: Error): boolean {
+    return /loading chunk [\d]+ failed/i.test(error.message);
   }
 
-  private handleOnClick() {
+  private handleToggleViewStack() {
     const expanded = !this.state.expanded;
     this.setState({
       expanded,
     });
   }
 
+  private handleReloadNow(): void {
+    window.location.reload();
+  }
+
+  private handleStopCountdown(): void {
+    this.toDispose.dispose();
+  }
+
+  private startCountdown(): void {
+    const id = window.setInterval(() => {
+      const reloadAfter = this.state.reloadAfter - 1;
+      this.setState({
+        reloadAfter,
+      });
+      if (reloadAfter === 0) {
+        this.handleReloadNow();
+      }
+    }, 1000);
+
+    this.toDispose.push({
+      dispose: () => window.clearInterval(id)
+    });
+  }
+
+  private buildErrorMessageAlert(): React.ReactNode {
+    const { error, errorInfo, expanded } = this.state;
+
+    const actionErrorTitle = expanded ? 'Hide stack' : 'View stack';
+    const errorName = error?.name ? error.name : Error;
+    const errorMessage = error?.message ? ': ' + error.message : '';
+
+    return (
+      <Alert
+        isInline
+        variant={AlertVariant.danger}
+        title={errorName + errorMessage}
+        actionLinks={
+          <AlertActionLink onClick={() => this.handleToggleViewStack()}>
+            {actionErrorTitle}
+          </AlertActionLink>
+        }
+      >
+        {expanded && errorInfo && (
+          <TextContent>
+            <Text component={TextVariants.pre}>
+              {errorInfo.componentStack}
+            </Text>
+          </TextContent>
+        )}
+      </Alert>
+    );
+  }
+
+  private buildReloadAlert(): React.ReactNode {
+    const { reloadAfter, shouldReload: willReload } = this.state;
+
+    if (willReload === false) {
+      return;
+    }
+
+    return (
+      <Alert
+        isInline
+        variant={AlertVariant.warning}
+        title={
+          <React.Fragment>
+            <span>This page is reloading in&nbsp;</span>
```

Comment:
I think we need to make it more clearer for the user why we're refreshing a page:
```
The application has been likely updated on the server. Refreshing a page to get newer resources in .
Contact an administrator if refreshing continues after the next load.
```
^ can we use local storage to detect when page is endless (2-3 times) refreshed automatically, to simplify the message for user and show the last part only after we detect it?

Reply:
sure, that's a really nice improvement